README.md

```diff
@@ -1,5 +1,37 @@
 # Rancher Catalog
 
+In Rancher, one of the two automatically packaged catalogs is this repository and the [community-catalog](https://github.com/rancher/community-catalog). The **library** catalog points to a git mirror of the **master** branch of this repo.
+
+## Branches
+
+When developing and testing new templates, a new dev branch for the upcoming release is created and packaged in our `rancher/server:master` container. For example, when we are developing and testing for `v1.6.0`, a `v1.6.0-dev` branch is created. Any PRs with changes should be made to this **dev** branch.
+
+Each Rancher RC is packaged with a specific branch to isolate what specific versions of templates were tested. If there are fixes that need to be made to address the templates, they should always be made to the **dev** branch. QA will pick up the **dev** branch for their RCs, if there are known issues in the specific branch packaged for the RC.
+
+> **Note:** Branches for specific RCs may be deleted in the future, as Rancher does not maintain the RCs.
+
+Before a Rancher release is created, a PR will be made to bring all changes from the **dev** branch to the **master** branch. The commits will be squashed into the 1 commit.
```

Comment:
FYI, I'd like to eventually get rid of branches of RCs and replace with tags, since we want them to be immutable snapshots anyway. But, all this is a discussion for later. This just documents the current process as it is today